tests/console/python3_pynacl.pm

```diff
@@ -0,0 +1,76 @@
+# SUSE's openQA tests
+#
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Simple pynacl libsodium smoke test with transactional and non-transactional support
+#
+# Maintainer: QE Core <qe-core@suse.de>
+#
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use serial_terminal 'select_serial_terminal';
+use registration qw(runtime_registration);
+use version_utils;
+use python_version_utils;
+
+my $requires_scc_registration = is_sle_micro || is_sle;
+my $script_download_path = "~/test_pynacl.py";
+
+sub run_test {
+    my ($python_package) = @_;    # Proper declaration of $python_package
+    my $pkg = "$python_package-PyNaCl";
+
+    # we don't run the test if PyNaCl is not packaged for this python version
+    return unless script_run("zypper se $pkg") == 0;
+
+    if (is_transactional) {
+        trup_call("pkg in $python_package $pkg");
+        process_reboot(expected_grub => 1, trigger => 1);
+        select_serial_terminal;
+    } else {
+        zypper_call("in $python_package $pkg");
+    }
+
+    my $python_interpreter = get_python3_binary($python_package);
+    record_info("running python version " . script_output("$python_interpreter --version"));
+
+    assert_script_run("$python_interpreter " . $script_download_path);
+
+    # clean up for the next run
+    if (is_transactional) {
+        trup_call("pkg rm $python_package $pkg");
+        process_reboot(expected_grub => 1, trigger => 1);
+        select_serial_terminal;
+    } else {
+        zypper_call("rm $python_package $pkg");
+    }
+}
+
+sub run {
+    my ($self) = @_;
+    select_serial_terminal;
+    # Download the test script
+    assert_script_run("curl -v -o $script_download_path " . data_url("python/pynacl/test_pynacl.py"));
+
+    runtime_registration() if $requires_scc_registration;
+    my @python3_versions = get_available_python_versions();
+    unshift @python3_versions, "python3";    # append the system default one
+    run_test($_) foreach @python3_versions;
+}
+
+sub post_fail_hook {
+    my $self = shift;
+    cleanup();
+    $self->SUPER::post_fail_hook;
+}
+
+sub cleanup {
+    remove_installed_pythons();
```

Comment:
This will not work on micro

Reply:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20411/files#diff-dd7bd57643b42f7bfc3114f8ee5c041f88154478b5060ef7e8a56b68e1b7bf24R88